components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java

```diff
@@ -521,6 +523,30 @@ public KeyManagerConfigurationDTO addKeyManagerConfiguration(
                     "Key manager Already Exist by Name " + keyManagerConfigurationDTO.getName() + " in tenant " +
                             keyManagerConfigurationDTO.getOrganization(), ExceptionCodes.KEY_MANAGER_ALREADY_EXIST);
         }
+
+        if (keyManagerConfigurationDTO.getAdditionalProperties().get(ISSUER_KEY) != null) {
+            try {
+                IdentityProvider idp = IdentityProviderManager.getInstance().getIdPByMetadataProperty(
+                        IDP_ISSUER_NAME,
+                        keyManagerConfigurationDTO.getAdditionalProperties().get(ISSUER_KEY).toString(),
+                        APIUtil.getInternalOrganizationDomain(keyManagerConfigurationDTO.getOrganization()),
+                        false);
+                if (idp != null) {
+                    throw new APIManagementException(
+                            String.format("Key Manager with issuer %s is already registered in tenant %s",
+                                    keyManagerConfigurationDTO.getAdditionalProperties().get(ISSUER_KEY).toString(),
+                                    keyManagerConfigurationDTO.getOrganization()), ExceptionCodes.DUPLICATE_ISSUER);
+                }
+            } catch (IdentityProviderManagementException e) {
+                throw new APIManagementException(String.format("IdP adding failed. %s", e.getMessage()),
+                        e,
+                        ExceptionCodes.IDP_ADDING_FAILED);
+            }
+        } else {
+            throw new APIManagementException("IdP adding failed. The issuer cannot be empty.",
+                    ExceptionCodes.IDP_ADDING_FAILED);
```

Comment:
Better to have a new Exception Code with "The issuer cannot be empty" as the description.

Reply:
ACK

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/11118/commits/c7402c0dac7b184bd56abb1513f4cad797ce0917